### PR TITLE
chore: [#186373445] re-enable FLP formation feature flag

### PR DESCRIPTION
**Pull request by seidior**
## Description

These code changes partially undo the work done in [#186335653](https://www.pivotaltracker.com/story/show/186335653) as the foreign limited partnership work is still under feature flag. With the acceptance of this pull request, the feature flag for foreign limited partnerships is restored as is a test to ensure the feature flag system is working.

### Ticket

[#186373445](https://www.pivotaltracker.com/story/show/186373445)

## Code author checklist

- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

File: web/src/lib/domain-logic/allowFormation.ts [hunk 1]
```diff
@@ -1,8 +1,29 @@
...
+  persona: BusinessPersona | undefined
+): boolean => {
+  const featureFlagMap: Partial<Record<FormationLegalType, boolean>> = {
+    "foreign-limited-partnership": process.env.FEATURE_BUSINESS_FLP === "true",
+  };
 
-export const allowFormation = (legalStructureId: string | undefined): boolean => {
   if (publicFilingLegalTypes.includes(legalStructureId as PublicFilingLegalType)) {
-    return true;
+    if (persona === "FOREIGN" || legalStructureId === "nonprofit") {
```
Comment (jphechter) [inline web/src/lib/domain-logic/allowFormation.ts:18]: [sand] nonprofits are no longer covered here. We can safely remove this portion of the check.
```suggestion
    if (persona === "FOREIGN") {
```

File: web/src/lib/roadmap/buildUserRoadmap.ts [hunk 2]
```diff
@@ -216,7 +216,7 @@ const getLegalStructureAddOns = (profileData: ProfileData): string[] => {
   if (profileData.legalStructureId === "nonprofit") {
     addOns.push("nonprofit");
 
-    const allowNonprofitFormation = allowFormation(profileData.legalStructureId);
+    const allowNonprofitFormation = allowFormation(profileData.legalStructureId, profileData.businessPersona);
```
Comment (jphechter) [inline web/src/lib/roadmap/buildUserRoadmap.ts:219]: [sand] I don't think we need this or the check on 220. All nonprofits (foreign and domestic) have formation enabled.

**Review by jphechter — APPROVED** [review]
One comment, but overall LGTM

File: web/src/lib/domain-logic/allowFormation.ts [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
+      return (
+        featureFlagMap[
+          castPublicFilingLegalTypeToFormationType(legalStructureId as PublicFilingLegalType, persona)
+        ] ?? true
+      );
+    } else {
+      return true;
+    }
```
Comment (aloverso) [inline web/src/lib/domain-logic/allowFormation.ts:26]: I know this is how it was before, but given the `?? true` on line 22, is the larger if-statement on line 18, and its `else true` necessary?

**Pull request merged**